packages/common/src/api/comments.ts

```diff
@@ -69,22 +85,57 @@ const commentsApi = createApi({
         return commentsRes
       },
       options: { type: 'mutation' },
-      async onQuerySuccess({ data: comment }, { entityId }, { dispatch }) {
-        dispatch(
-          commentsApi.util.updateQueryData(
-            'getCommentsByTrackId',
-            { entityId },
-            (prevState) => {
-              // TODO: how should we handle sorting here?
-              prevState?.unshift(comment) // add new comment to top of comment section
+      async onQuerySuccess(
+        { data: newId },
+        { entityId, body, userId, timestampS, parentCommentId },
+        { dispatch }
+      ) {
+        const newComment: Comment = {
+          id: newId,
+          userId,
+          message: body,
+          isPinned: false,
+          timestampS,
+          reactCount: 0,
+          replies: undefined,
+          createdAt: new Date().toISOString(),
+          updatedAt: undefined
+        }
+        optimisticUpdateCommentList(
+          entityId,
+          (prevState) => {
+            if (prevState) {
+              if (parentCommentId) {
+                const parentCommentIndex = prevState?.findIndex(
+                  (comment) => comment.id === parentCommentId
+                )
+                if (parentCommentIndex && parentCommentIndex >= 0) {
+                  const parentComment = prevState[parentCommentIndex]
+                  parentComment.replies = parentComment.replies || []
+                  parentComment.replies.push(newComment)
+                }
+              } else {
+                prevState.unshift(newComment) // add new comment to top of comment section
+              }
```

Comment:
Does this not have to return anything?

Reply:
No, the `prevState` is a proxy object so if you modify the existing one that works too. In this case its much easier to modify the existing one because spreading the proxy stuff is kind of weird. I saw some people online saying you can do `JSON.parse(JSON.stringify(proxyObj))` but I'm definitely not doing allat lol